src/utils/components/SidebarEditor/useEditorHighlighter.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useRef, useState } from 'react';
+
+import { YAMLEditorRef } from '@openshift-console/dynamic-plugin-sdk';
+
+import { createSelection, getLinesToHighlight } from './utils';
+
+export const useEditorHighlighter = (
+  editableYAML: string,
+  pathsToHighlight: string[],
+  showEditor: boolean,
+) => {
+  const [editor, setEditor] = useState<YAMLEditorRef['editor']>();
+  const isHighlighed = useRef(false);
```

Comment:
why do we need it? if you will use showEditor as your dependency at line 36 we will get same effect?